pkg/template/controller/templateinstance_controller.go

```diff
@@ -93,32 +109,77 @@ func (c *TemplateInstanceController) handle(templateInstance *templateapi.Templa
 		}
 	}
 
-	_, err = c.templateclient.TemplateInstances(templateInstance.Namespace).Update(templateInstance)
+	_, err = c.templateclient.TemplateInstances(templateInstance.Namespace).UpdateStatus(templateInstance)
 	if err != nil {
 		utilruntime.HandleError(fmt.Errorf("TemplateInstance update failed: %v", err))
 	}
 	return err
```

Comment:
this error is going to feed into a retry of the key, at which point what's going to happen? aren't we going to provision again, which will likely fail (duplicate objects) and then maybe we'll successfully update the template status condition to failed?  I think maybe we need to be more judicious in how we handle the failed update here (possibly retrying the update action right here and now, in some cases, so we don't go back through provision)